src/meta/src/hummock/compaction/picker/min_overlap_compaction_picker.rs

```diff
@@ -335,6 +356,35 @@ impl NonOverlapSubLevelPicker {
         }
 
         ret.sstable_infos.retain(|ssts| !ssts.is_empty());
+
+        // To check whether the task is expected
+        if ret.total_file_size > self.max_compaction_bytes
+            || ret.total_file_count as u64 > self.max_file_count
+            || ret.sstable_infos.len() > self.max_expected_level_count
+        {
+            // rotate the sstables to meet the `max_file_count` and `max_compaction_bytes` and `max_expected_level_count`
+            let mut total_file_count = 0;
+            let mut total_file_size = 0;
+            let mut total_level_count = 0;
+
+            for (index, sstables) in ret.sstable_infos.iter().enumerate() {
+                total_file_count += sstables.len();
+                total_file_size += sstables.iter().map(|sst| sst.file_size).sum::<u64>();
+                total_level_count += 1;
+
+                if index > 0
+                    && (total_file_count as u64 > self.max_file_count
+                        || total_file_size > self.max_compaction_bytes
+                        || total_level_count > self.max_expected_level_count)
+                {
+                    ret.total_file_count = total_file_count;
+                    ret.total_file_size = total_file_size;
+                    ret.sstable_infos.truncate(index);
+                    break;
```

Comment:
I do not undertand. 
Does it means that `total_file_count` does  represent the real number of files in this task ?

Reply:
yeah, this pr re-cut the compact task after picking. So its total_file_count and total_file_size will be changed